### ISSUE TITLE
allowing proxy adds undefined variable creation

### ISSUE THREAD

**Issue by FirefighterBlu3** [swig/swig#471]
from about 2012 to 2014 my project built fine with swig. i don't know what version it was, it is now 3.0.6. now, the same project code is seeing all the SwigMethods  basically duplicated but with pre/appended text which makes the variables undefined. this is intended as a python wrapper around libpcap.

example of what is correct:

``` c
{ (char *)"pcapObject_open_live", _wrap_pcapObject_open_live, METH_VARARGS, _doc_pcapObject_open_live },
```

example of new method entry that is not defined anywhere.

``` c
{ (char *)"pcapObject_open_live_doc_swigconstant", _wrap_pcapObject_open_live_doc_swigconstant, METH_VARARGS, _doc_pcapObject_open_live_doc_swigconstant },
```

``` c
pcap.c:3870:53: error: ‘_wrap_pcapObject_open_live_doc_swigconstant’ undeclared here (not in a function)
  { (char *)"pcapObject_open_live_doc_swigconstant", _wrap_pcapObject_open_live_doc_swigconstant, METH_VARARGS, _doc_pcapObject_open_live_doc_swigconstant },
                                                     ^
pcap.c:3870:112: error: ‘_doc_pcapObject_open_live_doc_swigconstant’ undeclared here (not in a function)
  { (char *)"pcapObject_open_live_doc_swigconstant", _wrap_pcapObject_open_live_doc_swigconstant, METH_VARARGS, _doc_pcapObject_open_live_doc_swigconstant },
                                                                                                                ^
```

all the methods are duplicated but come with an extra `_doc_swigconstant` on them. this full name is not found anywhere else in the project. the following exists, but does not have the `_wrap_` prefix.

``` c
SWIGINTERN PyObject *pcapObject_open_live_doc_swigconstant(PyObject *SWIGUNUSEDPARM(self), PyObject *args) {
```

given that there is a matching SWIGINTERN for `_wrap_pcapObject_open_live`, it looks like swig forgot to prefix the earlier SWIGINTERN with `_wrap_`

if i use -noproxy, none of these erroneous method entries are added to `SwigMethods[]`, but of course my proxy class is missing.

**Comment by vadz** [comment]
As always, it would be very helpful to have a [simple self-contained example](http://sscce.org/) reproducing the problem, could you please try to make one? Also, what command line options do you use?

**Comment by FirefighterBlu3** [comment]
i've been trying to create a working simplified example, not there yet.

swig -python -py3 -threads -o pcap.c pcap.i

**Comment by wsfulton** [comment]
@FirefighterBlu3  did you get a testcase together?

**Comment by ojwb** [comment]
There was a fix (0c819d1ba9a0eeb7c595bac86b74209ccf890052) to stop producing `_swigconstant` boilerplate for non-class types, so I wonder if this was fixed by that.

@FirefighterBlu3 Even if you can't produce a self-contained test case, could you retest with your full example?

**Comment by ojwb** [comment]
I'm going to close this on the assumption that the fix I linked to above has solved the problem.  The change which introduced the `_swigconstant` suffix is cfaf2f97fd207de1200eee821aab21c7d1014de3 which first appeared in v3.0.5 released 31 Jan 2015 which fits well with "from about 2012 to 2014 my project built fine with swig" and it not working with v3.0.6, while the candidate fix linked above is after the last indication that the problem still occurred.  It's also been more than 5 years since that candidate fix and we've not had another report of this problem since.

If you've seen this with SWIG 3.0.9 or later let us know and we can reopen.  As above, a simple self-contained reproducer would be very helpful.
